src/templates/edit.tsx

```diff
@@ -350,7 +368,11 @@ const Edit: () => JSX.Element = () => {
           />
         </DocumentProvider>
       ) : (
-        <LoadingScreen progress={progress} />
+        hasParent && typeof window !== "undefined" ?
```

Comment:
Clean this up a bit. You check window here but in the else case below you don't, but it's checked in the function.

Can you do something like:
```
useEffect(() => {
  if (typeof window !== "undefined" && !hasParent) {
    window.location.href = "/404.html";
  }
}, [hasParent]);
```

then here you keep this without the ternary. It would really be great if the window check was part of setting hasParent though so you don't have to keep checking it. Maybe we change to using a status with 3 values?

Reply:
for a similar reason we can't add the null window check to hasParent because then it will cause it to be false and immediately redirect, additionally the useEffect doesn't work here because we still need to check window not null and hasParent is true -- and i appears that having the window check in the if statement of the use effect leads to a hydration error

Reply:
done